Ntuplizer/plugins/NtupleBranches.cc

```diff
@@ -29,17 +29,27 @@ void NtupleBranches::branch( std::map< std::string, bool >& runFlags ){
       tree_->Branch( "genParticle_mass"	     , &genParticle_mass       ); 
       tree_->Branch( "genParticle_pdgId"     , &genParticle_pdgId      );
       tree_->Branch( "genParticle_status"    , &genParticle_status     );
+      tree_->Branch( "genParticle_isPrompt"  , &genParticle_isPrompt   );
+      tree_->Branch( "genParticle_isDirectPromptTauDecayProduct"  , &genParticle_isDirectPromptTauDecayProduct);
+      tree_->Branch( "genParticle_isDirectHardProcessTauDecayProductFinalState"  , &genParticle_isDirectHardProcessTauDecayProductFinalState);
+      tree_->Branch( "genParticle_fromHardProcessFinalState"  , &genParticle_fromHardProcessFinalState   );
       tree_->Branch( "genParticle_mother"    , &genParticle_mother     );
       tree_->Branch( "genParticle_nMoth"     , &genParticle_nMoth      );
       tree_->Branch( "genParticle_nDau"	     , &genParticle_nDau       ); 
       tree_->Branch( "genParticle_dau"	     , &genParticle_dau        );
+      tree_->Branch( "genParticle_taudau_pdgId"	     , &genParticle_taudau_pdgId        );
```

Comment:
This is only filled when we have taus with isPrompt flag is true. So, it won't change the total size at all and this will make our code super easier.